cli/pcluster/commands.py

```diff
@@ -48,20 +46,31 @@
 LOGGER = logging.getLogger(__name__)
 
 
-def _create_bucket_with_batch_resources(stack_name, resources_dir, region):
-    random_string = "".join(random.choice(string.ascii_lowercase + string.digits) for _ in range(16))
-    # bucket names must be at least 3 and no more than 63 characters long
-    s3_bucket_name = "-".join([stack_name.lower()[: 63 - len(random_string) - 1], random_string])
+def create_bucket_with_batch_resources(stack_name, region):
+    """
+    Create a bucket associated to the given stack and upload batch resources.
+
+    Returns the bucket name if both creation and upload succeed.
+    """
+    batch_resources = pkg_resources.resource_filename(__name__, "resources/batch")
+    s3_bucket_name = utils.generate_random_bucket_name(stack_name)
+    LOGGER.debug("Creating batch bucket named %s", s3_bucket_name)
 
     try:
-        utils.create_s3_bucket(bucket_name=s3_bucket_name, region=region)
-        utils.upload_resources_artifacts(bucket_name=s3_bucket_name, root=resources_dir)
-    except boto3.client("s3").exceptions.BucketAlreadyExists:
-        LOGGER.error("Bucket %s already exists. Please retry cluster creation.", s3_bucket_name)
+        utils.create_s3_bucket(s3_bucket_name, region)
+    except ClientError:
+        LOGGER.critical("Unable to create internal bucket %s. Please retry cluster creation.", s3_bucket_name)
         raise
+
+    try:
+        utils.upload_resources_artifacts(s3_bucket_name, root=batch_resources)
     except Exception:
-        utils.delete_s3_bucket(bucket_name=s3_bucket_name)
+        LOGGER.critical(
```

Comment:
Same here I would say

Minor: upload to the

Reply:
1. See previous comment for the exception management.
2. in --> to --> DONE